### PR TITLE
HumpFlickerFX version

### PR DESCRIPTION
**Pull request by NoSloppy**

### CODE REVIEW

File: functions/bump.h [hunk 1]
```diff
@@ -47,19 +47,27 @@ class Bump {
...
     pos_ = random(num_leds_);
+    // width_  = hump_width_.calculate(blade);
   }
   int getInteger(int led) {
-    return clampi32(abs(led - pos_) * 32768 / HUMP_WIDTH, 0, 32768);
+    // return clampi32(abs(led - pos_) * 32768 / width_, 0, 32768);
+    return clampi32(abs(led - pos_) * 32768 / hump_width_.getInteger(led), 0, 32768);
   }
 private:
+  // int width_;
```
Comment (profezzorn) [inline functions/bump.h:64]: remove commented out stuff

File: functions/bump.h [hunk 2]
```diff
@@ -47,19 +47,27 @@ class Bump {
...
 };
 
-template<int HUMP_WIDTH>
-class HumpFlickerF {
+template<class HUMP_WIDTH>
+class HumpFlickerFX {
 public:
   void run(BladeBase* blade) {
+    hump_width_.run(blade);
     int num_leds_ = blade->num_leds();
```
[the remark this answers is not in the document]
Comment (NoSloppy) [reply functions/bump.h]: ok. I haven't tested this change just fyi, but should be right now?.

File: functions/bump.h (hunk 1 above)
Comment (NoSloppy) [reply functions/bump.h]: oops

File: functions/bump.h [hunk 3]
```diff
@@ -47,19 +47,27 @@ class Bump {
   int mult_;
 };
 
-template<int HUMP_WIDTH>
-class HumpFlickerF {
+template<class HUMP_WIDTH>
```
[the remark this answers is not in the document]
Comment (NoSloppy) [reply functions/bump.h:60]: Please check the added usage for accuracy

[the remark this answers is not in the document]
Comment (profezzorn) [reply functions/bump.h:60]: Looks good to me.

**Pull request merged**